src/network/network_analyzer.rs

```diff
@@ -0,0 +1,86 @@
+use std::{
+	sync::{
+		atomic::{AtomicU32, Ordering},
+		Arc,
+	},
+	time::Duration,
+};
+
+use pcap::{ConnectionStatus, Device};
+use tokio::time;
+use tracing::{debug, info};
+
+// TODO: implement graceful shutdown for all async spawned functions
+pub async fn start_traffic_analyzer(port: u16, sampling_interval: u64) {
+	info!("Starting network analyzer.");
+	let devices = Device::list().unwrap();
+	let mut dev: Option<Device> = None;
+	for device in devices.clone() {
+		if !device.addresses.is_empty()
+		// The first interface with Connected status is usually the one with all the traffic
+			&& device.flags.connection_status == ConnectionStatus::Connected
+		{
+			dev = Some(device.clone());
+		}
+	}
+
+	debug!(
+		"Non lo device selected: {}",
+		dev.clone().expect("device should exist").name.as_str()
+	);
+
+	// listen to loopback device for local testing
+	let mut cap_loopback = pcap::Capture::from_device("lo")
+		.unwrap()
+		.immediate_mode(true)
+		.promisc(true)
+		.open()
+		.unwrap();
+
+	cap_loopback
+		.filter(&format!("udp port {port}"), true)
+		.unwrap();
+
+	let mut dev_cap =
+		pcap::Capture::from_device(dev.clone().expect("device should exist").name.as_str())
+			.unwrap()
+			.immediate_mode(true)
+			.promisc(true)
+			.open()
+			.unwrap();
```

Comment:
All unwraps from lines 38 through here cause panic.
Do we want that?

Reply:
Absolutely not, slipped my mind.